File: practice questions/Arrays/ReverseString.py
def reverse(arr: list) -> list:
    n = len(arr)
    start = 0
    end = n - 1
    while start < end:
        arr[start], arr[end] = arr[end], arr[start]
        start += 1
        end -= 1
    return arr

if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5, 6]
    print("Original array:", arr)
    reversed_arr = reverse(arr)
    print("Reversed array:", reversed_arr)

File: practice questions/Arrays/ReverseString.java
class ReverseString{
    public int[] reverse(int[] arr)
    {
        int n = arr.length;
        int[] rev = new int[n];
        int start =0 , end=n-1;
        while(start < n){
            rev[start]=arr[end];
            start++;
            end--;
        }
        return rev;
    }
    public static void main(String[] args) {
        int[] arr = { 1,2 ,3 ,4 ,5 ,6};
        ReverseString rs = new ReverseString();
        int[] res = rs.reverse(arr);
        for ( int i : res){
            System.out.print(i+" ");
        }
    }
}